### PR TITLE
Upgrade code-ready/machine, refactor libmachine

### PR DESCRIPTION
**Pull request by guillaumerose**
The main changes is the new logger. It is now using logrus.

Debug output:
```
time="2021-01-25T08:53:40+01:00" level=info msg="Creating CodeReady Containers VM for OpenShift 4.6.9..."
time="2021-01-25T08:53:40+01:00" level=debug msg="Found binary path at /home/guillaumerose/.crc/bin/crc-driver-libvirt"
time="2021-01-25T08:53:40+01:00" level=debug msg="Launching plugin server for driver libvirt"
time="2021-01-25T08:53:40+01:00" level=debug msg="Plugin server listening at address 127.0.0.1:34831"
time="2021-01-25T08:53:40+01:00" level=debug msg="() Calling .GetVersion"
time="2021-01-25T08:53:40+01:00" level=debug msg="Using API Version 1"
time="2021-01-25T08:53:40+01:00" level=debug msg="() Calling .SetConfigRaw"
time="2021-01-25T08:53:40+01:00" level=debug msg="() Calling .GetMachineName"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) Calling .GetMachineName"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) Calling .DriverName"
time="2021-01-25T08:53:40+01:00" level=debug msg="Running pre-create checks..."
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) Calling .PreCreateCheck"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) DBG | About to check libvirt version"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) DBG | Validating storage pool"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) Calling .GetConfigRaw"
time="2021-01-25T08:53:40+01:00" level=debug msg="Creating machine..."
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) Calling .Create"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) DBG | Preparing /home/guillaumerose/.crc/machines/crc/crc.qcow2 for machine use"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) DBG | image creation took 79.867012ms"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) DBG | Verifying executable bit set on /home/guillaumerose/.crc/machines/crc"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) DBG | Setting executable bit set on /home/guillaumerose/.crc/machines/crc"
time="2021-01-25T08:53:40+01:00" level=debug msg="(crc) DBG | Verifying executable bit set on /home/guillaumerose/.crc/machines"
```

Fixes https://github.com/code-ready/crc/issues/1886

### CODE REVIEW

File: cmd/crc/cmd/root.go [hunk 1]
```diff
@@ -160,12 +160,7 @@ func newMachine() machine.Client {
 }
 
 func newMachineWithConfig(config crcConfig.Storage) machine.Client {
-	return machine.NewClient(
-		constants.DefaultName,
-		isDebugLog(),
-		network.ParseMode(config.Get(cmdConfig.NetworkMode).AsString()),
-		config.Get(cmdConfig.EnableClusterMonitoring).AsBool(),
-	)
+	return machine.NewClient(constants.DefaultName, network.ParseMode(config.Get(cmdConfig.NetworkMode).AsString()), config.Get(cmdConfig.EnableClusterMonitoring).AsBool())
```
Comment (cfergeau) [inline cmd/crc/cmd/root.go:163]: almost tempted to write this as
```
    networkMode := network.ParseMode(config.Get(cmdConfig.NetworkMode).AsString())
    enableMonitoring := config.Get(cmdConfig.EnableClusterMonitoring).AsBool()
    machine.NewClient(constants.DefaultName, networkMode, enableMonitoring)
```
for readability.

**Review by cfergeau — COMMENTED** [review]
Some comments, but fairly straightforward changes.

File: pkg/libmachine/persist/filestore_test.go [hunk 2]
```diff
@@ -2,204 +2,132 @@ package persist
...
 	"github.com/code-ready/crc/pkg/libmachine/host"
-	"github.com/code-ready/crc/pkg/libmachine/hosttest"
+	"github.com/code-ready/crc/pkg/libmachine/version"
+	"github.com/stretchr/testify/assert"
 )
 
-func cleanup() {
-	os.RemoveAll(os.Getenv("MACHINE_STORAGE_PATH"))
-}
-
```
[the remark this answers is not in the document]
Comment (guillaumerose) [reply pkg/libmachine/persist/filestore_test.go:20]: done

File: pkg/libmachine/persist/filestore_test.go [hunk 3]
```diff
@@ -2,204 +2,132 @@ package persist
...
-	}
+	assert.True(t, ok)
 
 	realDriver := none.NewDriver(h.Name, store.MachinesDir)
+	assert.NoError(t, json.Unmarshal(rawDataDriver.Data, &realDriver))
+}
 
-	if err := json.Unmarshal(rawDataDriver.Data, &realDriver); err != nil {
-		t.Fatalf("Error unmarshaling rawDataDriver data into concrete 'none' driver: %s", err)
+func testHost() *host.Host {
```
[the remark this answers is not in the document]
Comment (guillaumerose) [reply pkg/libmachine/persist/filestore_test.go:125]: done

File: pkg/libmachine/host/host.go [hunk 4]
```diff
@@ -86,3 +87,18 @@ func (h *Host) UpdateConfig(rawConfig []byte) error {
 
 	return nil
 }
+
+func MachineInState(d drivers.Driver, desiredState state.State) func() error {
+	return func() error {
```
[the remark this answers is not in the document]
Comment (guillaumerose) [reply pkg/libmachine/host/host.go:92]: MachineInState is only used in crc now. After that, we can remove it from machine.

File: pkg/drivers/hyperv/hyperv.go [hunk 5]
```diff
@@ -158,11 +158,11 @@ func (d *Driver) Create() error {
 		if err != nil {
 			return err
 		}
-		log.Infof("Using switch %q", virtualSwitch)
+		log.Debugf("Using switch %q", virtualSwitch)
```
[the remark this answers is not in the document]
Comment (guillaumerose) [reply pkg/drivers/hyperv/hyperv.go:161]: Yes added comment + link to https://github.com/code-ready/crc/issues/1886

File: cmd/crc/cmd/root.go (hunk 1 above)
Comment (guillaumerose) [reply cmd/crc/cmd/root.go]: yes done

**Comment by guillaumerose** [comment]
All done except for the MachineInState change that is now only used here so we can adapt it.

**Pull request merged**